### PR TITLE
Authentication and authorization logging

### PR DESCRIPTION
**Pull request by g4s8**
#197 - added `INFO` logs for authentication and authorization operations.

### CODE REVIEW

File: src/main/java/com/artipie/PermissionsYaml.java [hunk 1]
```diff
@@ -36,7 +37,7 @@
  * on repository yaml configuration file.
  * @since 0.2
  */
-public final class RpPermissions implements Permissions {
+public final class PermissionsYaml implements Permissions {
```
Comment (olegmoz) [inline src/main/java/com/artipie/PermissionsYaml.java:40]: @g4s8 please consider renaming the class to `YamlPermissions` as it more common to put implementation specifics first and interface name second

File: src/main/java/com/artipie/PermissionsYaml.java [hunk 2]
```diff
@@ -52,23 +53,29 @@
...
     public boolean allowed(final String name, final String action) {
         final YamlMapping all = this.yaml.yamlMapping("permissions");
-        return check(all.yamlSequence(name), action)
-            || check(all.yamlSequence(RpPermissions.WILDCARD), action);
+        final boolean res = check(all.yamlSequence(name), action)
+            || check(all.yamlSequence(PermissionsYaml.WILDCARD), action);
+        if (res) {
+            Logger.info(this, "Operation '%s' allowed for '%s'", action, name);
+        } else {
+            Logger.info(this, "Operation '%s' denied for '%s'", action, name);
```
Comment (olegmoz) [inline src/main/java/com/artipie/PermissionsYaml.java:76]: @g4s8 logging level looks to be too high for this kind of messages as they explain they do not log actions, but explain reason why certain actions were made. I'd suggest to decrease logging level to `debug` or make it configurable

File: src/main/java/com/artipie/auth/AuthFromEnv.java [hunk 3]
```diff
@@ -71,8 +72,10 @@ public AuthFromEnv(final Map<String, String> env) {
         // @checkstyle LineLengthCheck (5 lines)
         if (Objects.equals(Objects.requireNonNull(username), this.env.get(AuthFromEnv.ENV_NAME))
             && Objects.equals(Objects.requireNonNull(password), this.env.get(AuthFromEnv.ENV_PASS))) {
+            Logger.info(this, "Successfully authenticated '%s' user", username);
             result = Optional.of(username);
         } else {
+            Logger.info(this, "Failed to authenticate '%s' user", username);
```
Comment (olegmoz) [inline src/main/java/com/artipie/auth/AuthFromEnv.java:78]: @g4s8 logging levels, see my previous comment

File: src/main/java/com/artipie/auth/AuthFromYaml.java [hunk 4]
```diff
@@ -84,6 +85,11 @@ public AuthFromYaml(final YamlMapping cred) {
                 }
             }
         }
+        if (res.isEmpty()) {
+            Logger.info(this, "Failed to authenticate '%s' user", user);
+        } else {
+            Logger.info(this, "Successfully authenticated '%s' user", user);
+        }
```
Comment (olegmoz) [inline src/main/java/com/artipie/auth/AuthFromYaml.java:92]: @g4s8 logging levels, see my previous comment

**Comment by 0crat** [comment]
This pull request [#271](https://github.com/artipie/artipie/issues/271) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/79d063e4-8142-4ef1-acba-603a83e2d9a0); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) I see that you unassigned this issue; the order is still assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk) though; to cancel the order use `refuse`, as in [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
This pull request [#271](https://github.com/artipie/artipie/issues/271) is assigned to @olegmoz[/z](https://www.0crat.com/u/olegmoz), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/f8008958-eb78-4d7e-a6a0-175a581e4195); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; @olegmoz[/z](https://www.0crat.com/u/olegmoz) is not a member of this project yet, but they can request to join, as [§1](http://www.zerocracy.com/policy.html#1) explains; there will be no monetary reward for this job

**Comment by g4s8** [comment]
@olegmoz thanks, should be fixed now, please see new changes

File: src/main/java/com/artipie/YamlPermissions.java [hunk 5]
```diff
@@ -52,23 +53,29 @@
...
         final YamlMapping all = this.yaml.yamlMapping("permissions");
-        return check(all.yamlSequence(name), action)
-            || check(all.yamlSequence(RpPermissions.WILDCARD), action);
+        final boolean res = check(all.yamlSequence(name), action)
+            || check(all.yamlSequence(YamlPermissions.WILDCARD), action);
+        if (res) {
+            Logger.info(this, "Operation '%s' allowed for '%s'", action, name);
+        } else {
+            Logger.info(this, "Operation '%s' denied for '%s'", action, name);
+        }
```
Comment (olegmoz) [inline src/main/java/com/artipie/YamlPermissions.java:77]: @g4s8 guess you forgot to delete logging here

**Comment by 0crat** [comment]
@olegmoz[/z](https://www.0crat.com/u/olegmoz) this job was assigned to you 6days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @olegmoz[/z](https://www.0crat.com/u/olegmoz), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**